packages/frontend/src/components/accounts/recovery_setup/new_account/SetupPassphraseNewAccount.js

```diff
@@ -0,0 +1,76 @@
+import { KeyPair } from 'near-api-js';
+import { generateSeedPhrase } from 'near-seed-phrase';
+import React, { useEffect, useState } from 'react';
+
+import { Mixpanel } from '../../../../mixpanel/index';
+import ConfirmPassphrase from './ConfirmPassphrase';
+import SavePassphrase from './SavePassphrase';
+
+export default ({ handleConfirmPassphrase }) => {
+    const [passPhrase, setPassPhrase] = useState('');
+    const [recoveryKeyPair, setRecoveryKeyPair] = useState();
+    const [implicitAccountId, setImplicitAccountId] = useState('');
+
+    const [confirmPassphrase, setConfirmPassphrase] = useState(false);
+    const [wordId, setWordId] = useState(null);
+    const [userInputValue, setUserInputValue] = useState('');
+    const [userInputValueWrongWord, setUserInputValueWrongWord] = useState(false);
+
+    const generateAndSetPhrase = () => {
```

Comment:
Not a blocker for this PR, just thinking out loud -- I think that everything in this function and the associated state that it mutates should be a custom hook so we can re-use it in the `SetupPassphraseExistingAccount` component when we create it :)